src/arrayMethodSort.js

```diff
@@ -4,8 +4,38 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let compareFn = compareFunction;
+
+    if (!compareFn) {
+      compareFn = (a, b) => {
+        const aStr = String(a);
+        const bStr = String(b);
+
+        if (aStr < bStr) {
+          return -1;
+        } else if (aStr > bStr) {
+          return 1;
+        }
+
+        return 0;
+      };
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - i - 1; j++) {
```

Comment:
The inner loop can be optimized by reducing the number of iterations. Since the largest element is guaranteed to be in its correct position after each iteration of the outer loop, you can subtract 'i' from the inner loop's condition: for (let j = 0; j < this.length - i - 1; j++).